### PR TITLE
handle UnloadFromMyGarages recipient address by PlanetID

### PR DESCRIPTION
**Pull request by ipdae**
- Fix FungibleAssetValue recipient address by ticker
- Handle recipient address by PlanetID
- Add memo for relay

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205715938209983

### CODE REVIEW

File: worker/worker/handler.py [hunk 1]
```diff
@@ -76,7 +80,8 @@ def process(sess: Session, message: SQSMessageRecord, nonce: int = None) -> Tupl
 
     unsigned_tx = gql.create_action(
         "unload_from_garage", pubkey=account.pubkey, nonce=nonce,
-        fav_data=fav_data, avatar_addr=message.body.get("avatar_addr"), item_data=item_data,
+        fav_data=fav_data, avatar_addr=avatar_address, item_data=item_data,
+        memo=f"agent_address: {agent_address}, avatar_address: {avatar_address}"
```
Comment (longfin) [inline worker/worker/handler.py:84]: It seems to be a little bit hard to parse. how about using simple json notation as like below...

```python
memo=json.dumps([agent_address, avatar_address])
```

File: worker/worker/handler.py [hunk 2]
```diff
@@ -61,13 +63,15 @@ def process(sess: Session, message: SQSMessageRecord, nonce: int = None) -> Tupl
...
-        }
-    } for x in product.fav_list]
+    planet_id: PlanetID = PlanetID(bytes(message.body["planet_id"], 'utf-8'))
+    agent_address = message.body.get("agent_addr")
+    avatar_address = message.body.get("avatar_addr")
+    # relay
+    if planet_id != PlanetID.ODIN:
+        vault_address = derive_vault_address(planet_id).hex()
+        agent_address = vault_address
+        avatar_address = vault_address
```
Comment (U-lis) [inline worker/worker/handler.py:73]: Is it OK to send agent-bound FAVs and avatar-bound inventory items to same address?
Because I don't know much about the `vault` and `relay`.

File: tests/utils/address_test.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import pytest
+
+from common.utils.address import format_addr, derive_vault_address
+from common.utils.receipt import PlanetID
```
Comment (U-lis) [inline tests/utils/address_test.py:4]: I cannot found `PlanetID` declaration in change list. Did I miss something?

**Review by U-lis — COMMENTED** [review]
Can you add `pytest tests` command into `.github/workflows/test.yml` to run pytest in CI process?

File: tests/utils/address_test.py (hunk 3 above)
Comment (ipdae) [reply tests/utils/address_test.py:4]: PlanetID added on #89

File: worker/worker/handler.py (hunk 2 above)
Comment (ipdae) [reply worker/worker/handler.py:73]: @longfin That's what I understand, is that correct?

File: common/utils/address.py [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
+        address = os.environ["ODIN_AVATAR_ADDRESS"]
+    elif planet_id == PlanetID.HEIMDALL:
+        address = os.environ["HEIMDALL_AVATAR_ADDRESS"]
+    elif planet_id == PlanetID.ODIN_INTERNAL:
+        address = os.environ["ODIN_INTERNAL_AVATAR_ADDRESS"]
+    elif planet_id == PlanetID.HEIMDALL_INTERNAL:
+        address = os.environ["HEIMDALL_INTERNAL_AVATAR_ADDRESS"]
+    else:
+        raise ValueError(f"{planet_id!r} is not a value {PlanetID}")
+    return format_addr(address)
```
Comment (longfin) [inline common/utils/address.py:40]:
```suggestion
    return format_addr(address)

```

**Review by longfin — APPROVED** [review]
